components/common/StyledText.tsx

```diff
@@ -7,34 +7,57 @@ import { typography } from '../../config/typography';
 type TextProps = {
   children: React.ReactNode;
   style?: StyleProp<TextStyle>;
+  extraSmall?: boolean;
   small?: boolean;
   title?: boolean;
   bold?: boolean;
   subtitle?: boolean;
+  regular?: boolean;
+  medium?: boolean;
+  semiBold?: boolean;
+  light?: boolean;
+  robotoMedium?: boolean;
+  rubikLight?: boolean;
+  colorLink?: boolean;
 };
 
 /**
  * @returns Text component
+ * @param extraSmall: boolean - extra small font size 10px
  * @param small: boolean - small font size 12px
  * @param title: boolean - title font size 18px and bold
  * @param subtitle: boolean - subtitle font size 16px and semi-bold
  * @param bold: boolean - bold font
+ * @param regular: boolean - regular font
+ * @param medium: boolean - medium font
+ * @param semiBold: boolean - semiBold font
+ * @param light: boolean - light font
+ * @param robotoMedium: boolean - roboto medium font
+ * @param rubikLight: boolean - rubik light font
  * @param style: StyleProp<TextStyle> - custom style
  * @param children: React.ReactNode - text
+ * @param colorLink: boolean
  */
+
 const StyledText = (props: TextProps & NativeText['props']) => {
-  const { style, small, title, subtitle, bold } = props;
+  const { style, extraSmall, small, title, subtitle, bold, regular, medium, semiBold, light, colorLink } = props;
 
   const { colors } = useTheme();
 
   const { primary } = typography;
 
   const textStyle = [
+    { color: colors.text },
+    extraSmall && { fontSize: 10 },
     small && { fontSize: 12 },
     title && { fontSize: 18, fontFamily: primary.bold },
     subtitle && { fontSize: 16, fontFamily: primary.semiBold },
     bold && { fontFamily: primary.bold },
-    { color: colors.text },
+    regular && { fontFamily: primary.regular },
+    medium && { fontFamily: primary.medium },
+    semiBold && { fontFamily: primary.semiBold },
+    light && { fontFamily: primary.light },
+    colorLink && { color: colors.link },
   ].filter(Boolean);
```

Comment:
The API of this component is becoming confusing. TypeScript should warn me if I try to do something like this `<StyledText subtitle title extraSmall>Text</StyledText>` since I am overriding the `fontSize` prop three times in the same component and it is not clear to the user of this component that `subtitle` will win since it is defined last in the `textStyle` array. Also all these "invalid" props (they are not valid props for the `Text` component) end up in the rendered JSX:

```jsx
<Text
  extraSmall={true}
  style={...}
  subtitle={true}
  title={true}
>
  Text
</Text>
```

The `props` and the JSDoc also include props like `robotoMedium` which have no visual effect on the component.

We should make this API better and ensure TypeScript only represents valid states of the component. Something in the line of:

```jsx
type StyledTextProps = {
  size?: 'xs' | 'sm' | 'md' | 'lg';
  font?: 'bold' | 'regular' | 'medium' | 'semiBold' | 'light';
  variant?: 'text' | 'link';
} & TextProps;

const StyledText = ({ style, size = 'md', font = 'regular', variant = 'text', ...rest }: StyledTextProps) => {
  const { colors } = useTheme();
  const { primary } = typography;

  const sizes: Record<NonNullable<StyledTextProps['size']>, number> = {
    xs: 10,
    sm: 12,
    md: 16,
    lg: 18,
  };
  const fonts: Record<NonNullable<StyledTextProps['font']>, string> = {
    bold: primary.bold,
    regular: primary.regular,
    medium: primary.medium,
    semiBold: primary.semiBold,
    light: primary.light,
  };
  const variants: Record<NonNullable<StyledTextProps['variant']>, string> = {
    text: colors.text,
    link: colors.link,
  };

  return (
    <NativeText
      style={StyleSheet.compose({ fontSize: sizes[size], fontFamily: fonts[font], color: variants[variant] }, style)}
      {...rest}
    />
  );
};
```

And we can split out a second small `Title` component that renders out a title and accepts an optional `subtitle` prop to render a sub title.

Reply:
ok, let me work on this.